lib/cloudex/cloudinary_api.ex

```diff
@@ -161,16 +181,16 @@ defmodule Cloudex.CloudinaryApi do
     }"
   end
 
-  @spec post(tuple | String.t(), binary, map) ::
+  @spec post(tuple | String.t(), binary, map, keyword) ::
           {:ok, %Cloudex.UploadedImage{}} | {:ok, %Cloudex.UploadedVideo{}} | {:error, any}
-  defp post(body, source, opts) do
-    with {:ok, raw_response} <- common_post(body, opts),
+  defp post(body, source, opts, poison_opts) do
+    with {:ok, raw_response} <- common_post(body, opts, poison_opts),
```

Comment:
so based on the comment above^, this could potentially be unchanged `common_post(body, opts)`